cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -287,12 +303,53 @@ private RestHighLevelClient getClient() {
 
       LOG.info("Create new Elasticsearch client for cluster {}", elasticHosts);
       HttpHost[] hosts = Arrays.stream(elasticHosts.split(",")).map(hostAndPort -> {
-        int pos = hostAndPort.indexOf(':');
-        String host = pos < 0 ? hostAndPort : hostAndPort.substring(0, pos);
-        int port = pos < 0 ? 9200 : Integer.parseInt(hostAndPort.substring(pos + 1));
-        return new HttpHost(host, port);
+        String scheme = "http";
+        String host = hostAndPort;
+
+        int schemeIdx = host.indexOf("://");
+        if (schemeIdx > 0) {
+          scheme = host.substring(0, schemeIdx);
+          host = host.substring(schemeIdx + 3);
+        }
+
+        int port;
+        int portIdx = host.lastIndexOf(":");
+        if (portIdx > 0) {
+          try {
+            port = Integer.parseInt(host.substring(portIdx + 1));
+          } catch (NumberFormatException var7) {
+            throw new IllegalArgumentException("Invalid HTTP host: " + host);
+          }
+
+          host = host.substring(0, portIdx);
+        } else {
+            port = scheme == "https" ? 9243 : 9200;
+        }
+
+        return new HttpHost(host, port, scheme);
       }).toArray(HttpHost[]::new);
-      this.client = client = new RestHighLevelClient(RestClient.builder(hosts));
+
+      RestClientBuilder builder = RestClient.builder(hosts);
+
+      builder.setHttpClientConfigCallback(httpClientConfigCallback -> {
+        if (this.credentialsUsername != null && this.credentialsPassword != null) {
+          LOG.info("Creating REST client with authentication, username: {}", this.credentialsUsername);
+          CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+          credentialsProvider.setCredentials(AuthScope.ANY,
+              new UsernamePasswordCredentials(this.credentialsUsername, this.credentialsPassword));
```

Comment:
Misalignment. We align the code as:

```java
credentialsProvider.setCredentials(AuthScope.ANY,
                                   new UsernamePasswordCredentials(this.credentialsUsername, this.credentialsPassword));
```
or
```java
```java
credentialsProvider.setCredentials(
  AuthScope.ANY, new UsernamePasswordCredentials(this.credentialsUsername, this.credentialsPassword));
```
```